src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,419 @@
+                    {
+                        XmlNode node = xnList[i];
+                        if (node.Name == "files")
+                        {
+                            ParseFile(node.ChildNodes, stInfo);
+                        }
+                        else if (node.Name == "methods")
+                        {
+                            ParseMethod(node.ChildNodes, stInfo);
+                        }
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static void ParseFile(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                foreach (XmlNode node in xn)
+                {
+                    if (stInfo.Document == node.Attributes["id"].Value)
+                    {
+                        stInfo.Filepath = node.Attributes["name"].Value;
+                        stInfo.Filename = Path.GetFileName(node.Attributes["name"].Value);
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static void ParseMethod(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                foreach (XmlNode node in xn)
+                {
+                    if (stInfo.Type == node.Attributes["containingType"].Value &&
+                        stInfo.Method == node.Attributes["name"].Value &&
+                        stInfo.Token == node.Attributes["token"].Value)
+                    {
+                        if (node.Attributes.Item(2).Name == "parameterNames")
+                        {
+                            stInfo.Param = node.Attributes["parameterNames"].Value;
+                        }
+                        ParseSequence(node.ChildNodes, stInfo);
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static void ParseSequence(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                foreach (XmlNode node in xn)
+                {
+                    if (node.Name == "sequencePoints")
+                    {
+                        ParseEntry(node.ChildNodes, stInfo);
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static int HexToInt(string value)
+        {
+            // strip the leading 0x
+            if (value.StartsWith("0x", StringComparison.OrdinalIgnoreCase))
+            {
+                value = value.Substring(2);
+            }
+            return Int32.Parse(value, NumberStyles.HexNumber);
+        }
```

Comment:
You can use [`Convert.ToInt64(string,int)`](https://docs.microsoft.com/en-us/dotnet/api/system.convert.toint64?view=net-5.0#System_Convert_ToInt64_System_String_System_Int32_) to convert with or without the `0x` prefix.

Reply:
I modified it.